src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.jsx

```diff
@@ -25,28 +27,34 @@ const CourseOutlineTray = ({ intl }) => {
 
   const { courseStatus } = useSelector(getCourseStatus);
   const activeSequenceId = useSelector(getSequenceId);
+  const { sections = {}, sequences = {} } = useSelector(getCourseOutline);
+  const courseOutlineStatus = useSelector(getCourseOutlineStatus);
 
   const {
     courseId,
+    unitId,
     currentSidebar,
     toggleSidebar,
     shouldDisplayFullScreen,
   } = useContext(SidebarContext);
 
-  const {
-    courseBlocks,
-  } = useModel('outline', courseId);
-
   const {
     sectionId: activeSectionId,
   } = useModel('sequences', activeSequenceId);
 
-  const rootCourseId = courseBlocks?.courses && Object.keys(courseBlocks?.courses)[0];
-  const sectionsIds = courseBlocks?.courses[rootCourseId]?.sectionIds;
-  const sequenceIds = courseBlocks?.sections[selectedSection || activeSectionId]?.sequenceIds;
-  const backButtonTitle = courseBlocks?.sections[selectedSection || activeSectionId]?.title;
-
-  const handleToggleCollapse = () => (currentSidebar === ID ? toggleSidebar(null) : toggleSidebar(ID));
+  const sectionsIds = Object.keys(sections);
+  const sequenceIds = sections[selectedSection || activeSectionId]?.sequenceIds || [];
+  const backButtonTitle = sections[selectedSection || activeSectionId]?.title;
+
+  const handleToggleCollapse = () => {
```

Comment:
is it okay that we have the same function here `CourseOutlineTray.jsx` and in the `CourseOutlineTrigger.jsx`?

Reply:
Yes, we need to handle trigger click the same from two places.